### DISCUSSION TITLE
[BUG] Conversations not restoring state when using auto-wire container

### DISCUSSION THREAD

**Discussion by joolab** [nutgram/nutgram#895] [Examples]
### Nutgram version

4.15

### PHP version

8.2

### Framework

No Framework

### Cache driver

Symfony Cache FilesystemAdapter

### OS

All

### Describe the bug

When using a Container with auto-wire support (e.g. PHP-DI) in the PHP library, Conversations stop working and the state is not restored. I suspect that the issue is deeper and may be related to the Hydrator. 

Upon analyzing the cache generated by Nutgram, it was found that when auto-wire is enabled, an incorrect key is created.

With auto-wire enabled:
```
1708016421
CONVERSATION_5617745145_5617745145_183321243
s:150:"O:28:"App\TgBot\Conversation\Admin":5:{s:12:"skipHandlers";b:0;s:15:"skipMiddlewares";b:0;s:4:"step";s:10:"uploadFile";s:6:"userId";N;s:6:"chatId";N;}";
```

With auto-wire disabled:
```
1708017550
CONVERSATION_5617745145_183321243_183321243
s:150:"O:28:"App\TgBot\Conversation\Admin":5:{s:12:"skipHandlers";b:0;s:15:"skipMiddlewares";b:0;s:4:"step";s:10:"uploadFile";s:6:"userId";N;s:6:"chatId";N;}";
```

### To Reproduce

To reproduce the issue, simply configure Nutgram using a Container with the auto-wire option enabled and try running a Conversation with multiple steps.

### Expected behavior

It is expected to work correctly with any DI Container, regardless of whether it implements auto-wire modes or not.

**Comment by sergix44** [comment]
Please take a look to this issue and see if that can be the case: https://github.com/nutgram/nutgram/issues/668 and this https://github.com/nutgram/nutgram/issues/468
Specifically for PHP-DI, i've experienced in the past weird non-psr-standard behaviour that might causing your issue, we have plenty of examples with symfony-di, laravel-container, league-container, and so on that works without issues. 
Feel free to reopen if needed.

**Comment by XAKEPEHOK** [comment]
For somebody, who (like me) needs to use PHP-DI with auto-wire i found a solution ( after 6 hours of research). Just make container wrapper for nutgram
```php
use Psr\Container\ContainerInterface;
use SergiX44\Container\Exception\NotFoundException;
use SergiX44\Nutgram\Conversations\Conversation;
use SergiX44\Nutgram\Telegram\Types\BaseType;

class NutgramContainer implements ContainerInterface
{

    //PHP-DI here
    public function __construct(private ContainerInterface $container)
    {
    }

    public function get(string $id)
    {
        if (!$this->container->has($id)) {
            throw new NotFoundException('Not found: ' . $id);
        }
        return $this->container->get($id);
    }

    public function has(string $id)
    {
        if (
            is_subclass_of($id, BaseType::class, true) ||
            is_subclass_of($id, Conversation::class, true) ||
            str_starts_with($id, 'SergiX44')
        ) {
            return false;
        }
        return $this->container->has($id);
    }
}
```

**Reply by jejestory02-rgb** [reply]
[]()t.me/@Scroll_[]()_bot_
